flocker/control/httpapi.py

```diff
@@ -397,9 +398,9 @@ def state_datasets(self):
         deployment = self.cluster_state_service.as_deployment()
         datasets = list(datasets_from_deployment(deployment))
         for dataset in datasets:
-            uuid = ip_to_uuid(dataset[u"primary"])
+            node_uuid = UUID(hex=dataset[u"primary"])
             dataset[u"path"] = self.cluster_state_service.manifestation_path(
-                uuid, dataset[u"dataset_id"]).path.decode(
```

Comment:
Maybe it'd be nice to update the parameter name too `manifestation_path(node_uuid...`